src/Spryker/Zed/EventBehavior/Business/Model/EventResourcePluginResolver.php

```diff
@@ -119,58 +85,62 @@ protected function mapPluginsByResourceName(): void
      */
     protected function getEffectivePlugins(array $resources): array
     {
-        $effectivePlugins = [];
         if (empty($resources)) {
             return $this->eventResourcePlugins;
         }
 
-        foreach ($resources as $resource) {
-            if (!isset($this->eventResourcePlugins[$resource])) {
-                throw new EventResourceNotFoundException(
-                    sprintf(
-                        'There is no resource with the name: %s.',
-                        $resource
-                    )
-                );
-            }
-
-            $effectivePlugins[$resource] = $this->eventResourcePlugins[$resource];
+        $filteredPlugins = array_filter($this->eventResourcePlugins, function (EventResourcePluginInterface $plugin) use ($resources) {
+            return in_array($plugin->getResourceName(), $resources);
+        });
+
+        if (!$filteredPlugins) {
```

Comment:
After refactoring, this method behaves weird, 
e.g. we have 3 resources [resource1,resource2,resource3]
The exception will be thrown only in case if for all 3 resources no plugins registered.
By I guess the intention was to throw an exception if plugin not registered for at least 1 of them.
Please verify

Reply:
Fixed, please re-check